### PR TITLE
Facebook login

### PR DESCRIPTION
**Pull request by felixp72**
Should be OK now.

Known issue: When using github login, if the user has his email marked as private, we don't get any email. And we should tell the user that in a nice way, not just behave like its an error on our side

### CODE REVIEW

File: backend/src/models/UsersModel.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import mongoose, { Schema, Document } from 'mongoose';
  * Stores all the users.
  */
 
-class UsersItem {
+export class UsersItem {
```
Comment (theodormoroianu) [inline backend/src/models/UsersModel.ts]: De ce dai export la clase?

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+function App(): JSX.Element {
   return (
     <Router basename={process.env.PUBLIC_URL}>
       <Switch>
         <Route key="default" path="/" exact component={DefaultPage}/>
         <Route key="testeval" path="/testeval" exact component={TestEval}/>
         <Route key="addgame" path="/addgame" exact component={AddGame}/>
+        <Route key="login" path="/login" exact component={Login}/>
+        <Route key="dashboard" path="/dashboard" exact component={Dashboard}/>
+        <Route key="smartregisterr" path="/smartregister" exact component={SmartRegister}/>
```
Comment (theodormoroianu) [inline frontend/src/App.tsx]: Nu exista o varianta mai usoara de-a face routarea?
Daca vrem de exemplu un element (cum ar fi title bar) care sa apara pe toate rutele, cum am face?

**Review by theodormoroianu — APPROVED** [review]

File: backend/src/models/UsersModel.ts (hunk 1 above)
Comment (felixp72) [reply backend/src/models/UsersModel.ts]: Bine zis, acum nu mai e nevoie

File: frontend/src/App.tsx (hunk 2 above)
Comment (felixp72) [reply frontend/src/App.tsx]: Nu stiu nici eu prea mult ce face chestia asta, deocamndata doar punem pagini random ce pot fi accesate in modul `<BASE_URL>:<PORT>/pagina`. Poate @ConstantinDragancea ne lumineaza putin

**Comment by felixp72** [comment]
This pr closes #31 

Currently known issues:
When using github, if the user didnt mark that his email is public, we dont get his email :D

**Comment by felixp72** [comment]
This PR introduces this new User story or bug, however you want to call it:  #59

File: frontend/src/App.tsx (hunk 2 above)
Comment (ConstantinDragancea) [reply frontend/src/App.tsx]: Tocmai de aia folosim Router. Switch functioneaza ca o componenta, deci putem avea chestii de genul
```
return (
  <>
     <Titlebar />
     <Switch>
        <Route ... />
        <Route ... />
     </Switch>
  </>
);
```
care va da render la ceva diferit in bucata de pagina corespunzatoare in dependenta de adresa la care ne aflam. Switch-ul trebuie incorporat intr-un Router (nu neaparat in aceeasi componenta, se poate afla si intr-un parinte al componentei) care sa-i spuna "radacina" switch-ului

**Pull request merged**